src/sgame/sg_struct.h

```diff
@@ -645,11 +645,29 @@ struct level_locals_s
 
 	bool     inClient;
 
-	int      time; // time the map was first started in milliseconds (map restart will update startTime)
+	// friend relations are to respectively setup m_startTime, m_time and to update m_previousTime
+	friend void G_InitGame( int, int, bool );
+	friend void G_RunFrame( int );
+
+	int time( void ) const { return m_time; }
+	void updateTime( int levelTime ){ m_time = levelTime; m_matchTime = levelTime - m_startTime; }
+	int startTime( void ) const { return m_startTime; }
+	int matchTime( void ) const { return m_matchTime; }
+
+	int lastFrameDelay( void ) const { return m_time - previousTime; }
+	int ageMinutes( void ) const { return ( m_time - m_startTime ) / 60000; }
+	int ageSeconds( void ) const { return ( m_time - m_startTime ) / 1000; }
+	bool warmingUp( void ) const;
```

Comment:
+1 for making this a function, this can reduce code duplication here and there. For the record, I think I had to duplicate the condition for the juggernaut mod

Reply:
That kind of thing should be a function e.g. `int G_MatchTimeSeconds()` for stylistic consistency with existing code.